controllers/infinispan_controller.go

```diff
@@ -301,22 +301,25 @@ func (reconciler *InfinispanReconciler) Reconcile(ctx context.Context, ctrlReque
 
 	// If an overlay configuration is specified wait for the related ConfigMap
 	overlayConfigMap := &corev1.ConfigMap{}
-	var overlayConfigMapKey string
+	var overlayConfigMapKey, overlayLog4jConfigMapKey string
 	if infinispan.Spec.ConfigMapName != "" {
 		if result, err := kube.LookupResource(infinispan.Spec.ConfigMapName, infinispan.Namespace, overlayConfigMap, infinispan, r.Client, reqLogger, r.eventRec, r.ctx); result != nil {
 			return *result, err
 		}
-		var foundKey bool
-		// Loop through the data looking for something like xml,json or yaml
+		// Loop through the data looking for something like xml, json or yaml
 		for overlayConfigMapKey = range overlayConfigMap.Data {
 			if overlayConfigMapKey == "infinispan-config.xml" || overlayConfigMapKey == "infinispan-config.json" || overlayConfigMapKey == "infinispan-config.yaml" {
-				foundKey = true
 				break
 			}
 		}
-		if !foundKey {
-			err := fmt.Errorf("infinispan-copnfig.[xml|yaml|json] configuration not found in ConfigMap: %s", overlayConfigMap.Name)
+		// Check if the user added a custom log4j.xml config
+		_, logFileExists := overlayConfigMap.Data["log4j.xml"]
+
+		if overlayConfigMapKey == "" && !logFileExists {
```

Comment:
This confuses me. If I'm not wrong `overlayConfigMapKey` will be the last key in the `.Data` map when the `for` loops over the whole map.

also is `&&` the correct boolean op here? I mean if logFileExists and a configuration is not found the `else` branch is executed

Reply:
great observation @rigazilla! I'm currently updating it and will push it shortly, thanks!